osquery/registry/registry_interface.cpp

```diff
@@ -13,6 +13,8 @@
 
 namespace osquery {
 void RegistryInterface::remove(const std::string& item_name) {
+  WriteLock lock(mutex_);
```

Comment:
You are holding a `ReadLock` in setUp and then calling `::remove`, which will try to grab a secondary `WriteLock` and deadlock.

Reply:
One way of handling this is to create a `::removeUnsafe` method with the critical section, then provide the `::remove` public method that grabs the lock and waterfalls into `::removeUnsafe`. We'd have to look at the callsites of `::remove` (outside of this implementation) and see if that is OK.

Reply:
Yeah that makes sense. Implemented a private `RegistryInterface::removeUnsafe` method with those changes.